crates/bevy_window/src/window.rs

```diff
@@ -90,6 +93,13 @@ impl WindowResizeConstraints {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct WindowIcon {
```

Comment:
Should there be validation here for `width * height == bytes.len()`?

Reply:
Should be `width * height * 4 == bytes.len()`,no? 
Where would you put it? In the Constructor for WindowIcon?

Reply:
> Should be width * height * 4 == bytes.len(),no?

Perhaps? I was just guessing haha 😋

> Where would you put it? In the Constructor for WindowIcon?

Yep!
I would do something like:
```rust
impl WindowIcon {
    fn new(bytes: Vec<u8>, width: usize, height: usize) -> Option<Self> 
    { /*..*/ }
}
```

Reply:
Oh actually probably just combine the width/height into a `bevy::math::UVec2`

Reply:
What benefit does this have? 
And how does it relate to your initial comment?

Reply:
When you create an icon view `new` the last two parameters are not strong custom types. 
This means it's possible for a user to accidentally swap those two parameters. 

Using a `UVec2` or something like:
```rust
struct Extents {
    width: u32,
    height: u32,
}
```
it would possibly help prevent a class of user errors (theoretically). 

However, in retrospect, I think I'm ok with the two `u32`s directly as well.....


In doesn't really relate to my first comment other than how you would calculate the dimensions
```rust
vec.x() * vec.y() * 4 == bytes.len()
```

Reply:
I dont like UVec2. I could create something like Extend2d, but I would not be able to put it into bevy_render :D.
So I will keep it like it is for now.

Regarding the Error Handling:
This is already handled in winit, should we replicate the checks for our icon type? In that case we should also return an error not an option.
Question about texture: Is it always rgba8? So that both the checks are always fulfilled for textures?

Reply:
I recreated the error and for now I do not assume that texture will necessarily be fine to use.